src/DotNet.Status.Web/Controllers/AzurePipelinesController.cs

```diff
@@ -233,31 +233,77 @@ private async Task ProcessBuildNotificationsAsync(Build build)
 
 {changesMessage}
 ";
+                    string issueTitlePrefix = $"Build failed: {build.Definition.Name}/{prettyBranch} {prettyTags}";
+                    bool updateExisting = repo.UpdateExisting;
 
-                    var newIssue =
-                        new NewIssue($"Build failed: {build.Definition.Name}/{prettyBranch} #{build.BuildNumber}")
-                        {
-                            Body = body,
+                    if (updateExisting)
+                    {
+                        RepositoryIssueRequest issueRequest = new RepositoryIssueRequest {
+                            Creator = "dotnet-eng-status[bot]",
+                            State = ItemStateFilter.Open,
+                            SortProperty = IssueSort.Created,
+                            SortDirection = SortDirection.Descending
                         };
 
-                    if (!string.IsNullOrEmpty(monitor.Assignee))
-                    {
-                        newIssue.Assignees.Add(monitor.Assignee);
-                    }
-                    
-                    foreach (string label in repo.Labels.OrEmpty())
-                    {
-                        newIssue.Labels.Add(label);
+                        if (!string.IsNullOrEmpty(monitor.Assignee))
+                        {
+                            issueRequest.Assignee = monitor.Assignee;
```

Comment:
assignee is not necessarily a constant so I don't know that we should search on it.